client/src/components/dataset/ColumnHeader.jsx

```diff
@@ -119,11 +129,43 @@ export default class ColumnHeader extends Component {
   }
 }
 
+const ConditionalTooltip = ({ children }) => {
+  const [useTooltip, setUseTooltip] = useState(false); // eslint-disable-line
+  const handleRef = (ref) => {
+    if (!ref) return;
+    if (ref.parentElement.offsetWidth < ref.offsetWidth + 40 && !useTooltip) {
+      setUseTooltip(true);
+    } else if (ref.parentElement.offsetWidth >= ref.offsetWidth + 40 && useTooltip) {
+      setUseTooltip(false);
+    }
+  };
```

Comment:
Why not `useRef`?

Reply:
There is no need for it in this case. A ref handler function is sufficient.

Reply:
Okay 👍🏾